patches/0100-Add-OpenSSL-crypto-module.patch

```diff
@@ -90,10 +110,22 @@ index 0000000000..4de8d404f8
 +	"crypto/internal/boring/sig"
 +	"hash"
 +	"math/big"
++	"runtime"
++	"syscall"
 +)
 +
 +const Enabled = false
 +
++func init() {
++	if v, ok := syscall.Getenv("GOFIPS"); ok && v == "1" {
```

Comment:
We still handle `GOLANG_FIPS` in the OpenSSL backend (and might actually end up changing to only support it, with https://github.com/microsoft/go/issues/397), should this have an equivalent check for `GOLANG_FIPS=1` too? Maybe this could be unified with the implementation on the OpenSSL backend by moving the env checks to a shared file?

https://github.com/microsoft/go/blob/a34208fdb5c5d337bd245671eeabb791c94a9746/patches/0100-Add-OpenSSL-crypto-module.patch#L286-L292

Reply:
Done! Environment variable retrieval is now in a common file.